lib/Result.ts

```diff
@@ -435,35 +420,36 @@ export const promiseResult = <Success, Failure>(
   promise: Promise<Result<Success, Failure>> | Result<Success, Failure>
 ): PromiseResult<Success, Failure> => {
   if (promise instanceof PromiseResult) {
-    return promise
+    return promise;
   }
   if (promise instanceof Promise) {
-    Object.setPrototypeOf(promise, PromiseResult.prototype)
-    return promise
+    Object.setPrototypeOf(promise, PromiseResult.prototype);
+    return promise as PromiseResult<Success, Failure>;
   }
-  return PromiseResult.resolve(promise)
-}
+
+  return new PromiseResult<Success, Failure>((resolve: any) => {
+    resolve(promise);
+  });
```

Comment:
Is there an issue with just using `PromiseResult.resolve`? If it's prototype based I suppose we could simply use the `Object.setPrototypeOf` trick.

Reply:
that should work, ty